Library/brew.rb

```diff
@@ -39,13 +38,24 @@
   EOS
 end
 
+# Check for user agreement of the Xcode license before permitting
+# any other brew usage to continue. This prevents the situation where
+# people are instructed to "please re-run as root via sudo" on brew commands.
+# The check can only fail when Xcode is installed & the active developer dir.
+if `/usr/bin/xcrun clang 2>&1` =~ /license/ && !$?.success?
+  odie <<-EOS.undent
+    You have not agreed to the Xcode license. Please resolve this by running:
+      sudo xcodebuild -license
+  EOS
+end
```

Comment:
This happens:
```
xcode-select: note: no developer tools were found at '/Applications/Xcode.app', requesting install. Choose an option in the dialog to download the command line developer tools.
```
We could throw it inside an Xcode existing block though, easily enough, at the cost of a little extra time. 

I deliberately didn't put it in `brew doctor` or another subcommand because if you haven't agreed to the license this happens:
```
Vagrants-Mac:local vagrant$ brew doctor
Agreeing to the Xcode/iOS license requires admin privileges, please re-run as root via sudo.
Error: Failure while executing: /usr/bin/otool -L /usr/bin/install_name_tool
```
```
Vagrants-Mac:local vagrant$ brew install -fs libtool
Agreeing to the Xcode/iOS license requires admin privileges, please re-run as root via sudo.
Error: Failure while executing: /usr/bin/otool -L /usr/bin/install_name_tool
```
Which implies the `brew` command should be run as root, and we've seen a lot of people hitting this and being confused by it lately, which is why I kind of wanted to put it in `brew` so they literally couldn't run a brew command without handling the license situation first.